src/data/nat/prime_factorization.lean

```diff
@@ -0,0 +1,82 @@
+/-
+Copyright (c) 2021 Stuart Presnell. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stuart Presnell
+-/
+import data.nat.prime
+import data.nat.mul_ind
+
+/-!
+# Prime factorizations
+
+ `n.prime_factorization` is the finitely supported function `ℕ →₀ ℕ`
+ mapping each prime factor of `n` to its multiplicity in `n`.  For example, since 2000 = 2^4 * 5^3,
+  * `prime_factorization 2000 2` is 4
+  * `prime_factorization 2000 5` is 3
+  * `prime_factorization 2000 k` is 0 for all other `k : ℕ`.
+
+-/
+
+open nat finset list finsupp
+open_locale big_operators
+
+namespace nat
+
+/-- `n.prime_factorization` is the finitely supported function `ℕ →₀ ℕ`
+ mapping each prime factor of `n` to its multiplicity in `n`. -/
+noncomputable def prime_factorization (n : ℕ) : ℕ →₀ ℕ := (n.factors : multiset ℕ).to_finsupp
```

Comment:
This is a bit long of a name. Feel free to shorten it to `prime_fact`, `prime_fac` or `factorization` (after all, it's `nat.factors`, not `nat.prime_factors`).

Same goes for the filename.